sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -324,7 +324,8 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
       DescribeTableCommand(
         visitTableIdentifier(ctx.tableIdentifier),
         partitionSpec,
-        ctx.EXTENDED != null || ctx.FORMATTED != null)
+        ctx.EXTENDED != null,
+        ctx.FORMATTED != null)
```

Comment:
we deprecated `DESC FORMATTED` intentionally, and made `DESC EXTENDED` behave as `DESC FORMATTED`. I really don't think users will query the result of `DESC EXTENDED` before, as all the detail table information is inside one column, so it's probably fine to break this behavior just for display.